includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -3331,4 +3334,32 @@ public function is_payment_request_enabled() {
 		return in_array( WC_Stripe_Payment_Methods::APPLE_PAY, $enabled_payment_method_ids, true ) ||
 			in_array( WC_Stripe_Payment_Methods::GOOGLE_PAY, $enabled_payment_method_ids, true );
 	}
+
+	/**
+	 * Maybe deactivate Affirm or Klarna payment methods if other official plugins are active.
+	 *
+	 * @return void
+	 */
+	private function maybe_deactivate_bnpls() {
+		if ( WC_Stripe_Helper::has_other_bnpl_plugins_active() ) {
+			$enabled_payment_methods       = $this->get_upe_enabled_payment_method_ids();
+			$payment_method_ids_to_disable = [];
+			if ( in_array( WC_Stripe_Payment_Methods::AFFIRM, $enabled_payment_methods, true )
+				&& WC_Stripe_Helper::has_gateway_plugin_active( WC_Stripe_Helper::OFFICIAL_PLUGIN_ID_AFFIRM ) ) {
+				$payment_method_ids_to_disable[] = WC_Stripe_Payment_Methods::AFFIRM;
+			}
```

Comment:
Should we also require that the relevant payment gateway is enabled before we actively make the API call to Stripe?  Given that we're doing this in the constructor, we might trip over a number of tricky race conditions if that's part of the behaviour. But maybe it would be worth hooking into a much later hook to handle this cleanup anyway?

Reply:
> Should we also require that the relevant payment gateway is enabled before we actively make the API call to Stripe?

`WC_Stripe_Helper::has_gateway_plugin_active` already checks if the gateway is enabled 👍 

> But maybe it would be worth hooking into a much later hook to handle this cleanup anyway?

Any suggestions? I could not find a better place to implement this call 🤔

Reply:
Good suggestion, but I went ahead and used `init` instead to follow the pattern we have in the main `WC_Stripe` class (moved the method there). Per the documentation above, it is close to `wp_loaded` (occurring just before).